### PR TITLE
chore: use a real connection in dao test

### PR DESCRIPTION
**Pull request by jcosentino11**
*Issue #, if available:*

*Description of changes:*
Rather than mock db connections, use a real one.  `DiskSpoolDAOTest` will now spin up a real sqllite db, initialize the tables, and perform the test.  

In order to test failure scenarios, such as throwing an exception when executing a statement, wrapper classes `ConnectionFake` and `StatementFake` have been made.  Test cases can then configure the fakes to exhibit special behavior, such as throw an exception on `PreparedStatement#executeUpdate`

```
SQLException sqlException = new SQLException("DB is corrupt", "some state", 11);
dao.getConnection().addExceptionOnUpdate(sqlException);
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/aws/greengrass/disk/spool/ConnectionFake.java [hunk 1]
```diff
@@ -0,0 +1,1912 @@
+/*
```
Comment (MikeDombo) [inline src/test/java/com/aws/greengrass/disk/spool/ConnectionFake.java:1]: Do we need this whole class? It is pretty insane...

Comment (MikeDombo) [reply src/test/java/com/aws/greengrass/disk/spool/ConnectionFake.java:1]: You may find lombok's delegate annotation helpful.

Comment (jcosentino11) [reply src/test/java/com/aws/greengrass/disk/spool/ConnectionFake.java:1]: Yeah I felt insane writing it.... that annotation looks very promising, taking a look

Comment (jcosentino11) [reply src/test/java/com/aws/greengrass/disk/spool/ConnectionFake.java:1]: Dropped it by 1k lines so that's good :P

**Review by MikeDombo — DISMISSED** [review]

**Review by MikeDombo — APPROVED** [review]

**Pull request merged**